crates/chain/src/tx_graph.rs

```diff
@@ -1180,56 +1502,98 @@ impl<A: Ord> Merge for ChangeSet<A> {
     }
 }
 
-impl<A: Ord> ChangeSet<A> {
+/// The [`ChangeSet`] represents changes to a [`TxGraph`] and it's internal [`Indexer`].
+///
+/// Refer to the [module-level documentation](crate::tx_graph) for more.
+///
+/// [`TxGraph`]: crate::TxGraph
+/// [`Indexer`]: crate::Indexer
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize, serde::Serialize),
+    serde(bound(
+        deserialize = "A: Ord + serde::Deserialize<'de>, XC: serde::Deserialize<'de>",
+        serialize = "A: Ord + serde::Serialize, XC: serde::Serialize",
+    ))
+)]
+#[must_use]
+pub struct ChangeSet<A = (), XC = ()> {
```

Comment:
Maybe we could just call this `I`, but either way is fine.
```suggestion
pub struct ChangeSet<A = (), I = ()> {
```